services/TemplateService.go

```diff
@@ -58,6 +58,30 @@ func (ts *TemplateService) Delete(ctx context.Context, template resources.Templa
 	return err
 }
 
+// Instantiate instantiates a new virtual machine from a template.
+func (ts *TemplateService) Instantiate(ctx context.Context, template resources.Template, name string,
+	onHold bool, blueprint blueprint.Interface, withImage bool) (*resources.VirtualMachine, error) {
```

Comment:
I would call `withImage` parameter something different. Maybe `persistent`, that's how it's called in Sunstone after all.